### PR TITLE
table/pagination-changes

### PR DESCRIPTION
**Pull request by RichardX366**
Made total pages for pagination unrequired, added a pagination option to table, and added a no data property along with a name property to customize.

### CODE REVIEW

File: src/components/Table/index.tsx [hunk 1]
```diff
@@ -1,42 +1,88 @@
...
+  pagination?: PaginationProps;
 }
 
 export const Table: React.FC<TableProps> = ({
   stripes,
   columns,
-  data
+  data,
+  name,
+  noDataPlaceholder = (
```
Comment (mxdi9i7) [inline src/components/Table/index.tsx:19]: can we reuse the current placeholder in OA? i kinda like that better

File: src/components/Table/index.tsx [hunk 2]
```diff
@@ -1,42 +1,88 @@
...
-            <tbody>
-            {data.map((row, i) => (
-              <tr key={row.id} className={!stripes || i % 2 === 0 ? 'bg-white' : 'bg-gray-50'}>
-                {columns.map(column => (
-                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{row[column]}</td>
+              <tr>
+                {columns.map((column) => (
+                  <th
+                    scope="col"
+                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
```
Comment (mxdi9i7) [inline src/components/Table/index.tsx:43]: key

File: src/components/Table/index.tsx [hunk 3]
```diff
@@ -1,42 +1,88 @@
...
+              {data.length ? (
+                data.map((row, i) => (
+                  <tr
+                    key={row.id}
+                    className={
+                      !stripes || i % 2 === 0 ? 'bg-white' : 'bg-gray-50'
+                    }
+                  >
+                    {columns.map((column) => (
+                      <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
```
Comment (mxdi9i7) [inline src/components/Table/index.tsx]: key

File: src/components/Table/index.tsx [hunk 4]
```diff
@@ -1,42 +1,88 @@
...
+                ))
+              ) : (
+                <tr>
+                  <td className="bg-white p-6" colSpan={columns.length}>
+                    {noDataPlaceholder}
+                  </td>
+                </tr>
+              )}
             </tbody>
+            {pagination && (
```
Comment (mxdi9i7) [inline src/components/Table/index.tsx]: I'd expect pagination to be a required prop, there's no table in the world that doesn't paginate

**Review by mxdi9i7 — CHANGES_REQUESTED** [review]

File: src/components/Table/index.tsx (hunk 4 above)
Comment (RichardX366) [reply src/components/Table/index.tsx]: What about students in a cohort? All students in it are fetched when the cohort is fetched in cohort detail.

Comment (mxdi9i7) [reply src/components/Table/index.tsx]: there could be more students than 1 page tho

**Review by mxdi9i7 — APPROVED** [review]

**Pull request merged**